packages/sommelier.rb

```diff
@@ -2,106 +2,256 @@
-      system "echo 'fi' >> stopsommelier"
-      system "echo 'SOMM=\$(pidof sommelier 2> /dev/null)' >> stopsommelier"
-      system "echo 'if [ -z \"\$SOMM\" ]; then' >> stopsommelier"
-      system "echo '  echo \"sommelier process stopped\"' >> stopsommelier"
-      system "echo 'else' >> stopsommelier"
-      system "echo '  echo \"sommelier process \$SOMM is running\"' >> stopsommelier"
-      system "echo '  exit 1' >> stopsommelier"
-      system "echo 'fi' >> stopsommelier"
-      system "echo '#!/bin/bash' > startsommelier"
-      system "echo 'set -a && source ~/.sommelier.env && set +a && initsommelier' >> startsommelier"
+    # There is no good way to checksum the googlesource tgz file, as they appear to be generated on the fly
+    # and checksums vary with each download.
+    system 'curl -L https://chromium.googlesource.com/chromiumos/platform2/+archive/c6f85a3472584df37fcf2d3d99474081f29256cd.tar.gz | tar mzx --warning=no-timestamp'
+    Dir.chdir ("vm_tools/sommelier") do
+    
+    # Google's sommelier expects to find virtwl.h in their kernel source includes, but we may not have
+    # set of kernel headers which match, so we just download virtwl.h and then patch the sommelier source
+    # to look for the file locally.
+    ######################### Download virtwl.h from Chromium 5.4 kernel tree ###########################################
+    url_virtwl = "https://chromium.googlesource.com/chromiumos/third_party/kernel/+/5d641a7b7b64664230d2fd2aa1e74dd792b8b7bf/include/uapi/linux/virtwl.h?format=TEXT"
+    uri_virtwl = URI.parse url_virtwl
+    #filename_virtwl = File.basename(uri_virtwl.path)
+    filename_virtwl = 'virtwl.h_base64'
+    sha256sum_virtwl = 'a8215f4946ccf30cbd61fcf2ecc4edfe6d05bffeee0bacadd910455274955446'
+    
+    if File.exist?(filename_virtwl) && Digest::SHA256.hexdigest( File.read("./#{filename_virtwl}") ) == sha256sum_virtwl
+      puts "Unpacking virtwl source code".yellow
+    else
+      puts "Downloading virtwl".yellow
+      system('curl', '-s', '-C', '-', '--insecure', '-L', '-#', url_virtwl, '-o', filename_virtwl)
+      abort 'Checksum mismatch. :/ Try again.'.lightred unless
+        Digest::SHA256.hexdigest( File.read("./#{filename_virtwl}") ) == sha256sum_virtwl
+      puts "virtwl archive downloaded".lightgreen
+      system 'mkdir -p build/linux'
+      system 'base64 --decode virtwl.h_base64 > build/linux/virtwl.h'
+    end    
+    
+    # lld is needed so libraries linked to system libraries (e.g. libgbm.so) can be linked against, since those are required for graphics acceleration.
+    ENV['CFLAGS'] = "-fuse-ld=lld"
+    ENV['CXXFLAGS'] = "-fuse-ld=lld"
+    
+    system "meson #{CREW_MESON_OPTIONS} -Dxwayland_path=#{CREW_PREFIX}/bin/Xwayland -Dxwayland_gl_driver_path=/usr/#{ARCH_LIB}/dri -Ddefault_library=both -Dxwayland_shm_driver=noop -Dshm_driver=noop -Dvirtwl_device=/dev/null -Dpeer_cmd_prefix=\"#{CREW_PREFIX}#{PEER_CMD_PREFIX}\" build"
```

Comment:
Can we split this onto separate lines for readability?

Reply:
done